src/eg/src/election_record.rs

```diff
@@ -34,20 +40,30 @@ pub struct PreVotingData {
     /// The joint election public key.
     pub public_key: JointElectionPublicKey,
 }
+
 #[allow(dead_code)]
 /// The body of the election record, generated after the election is complete.
 #[derive(Debug)]
 pub struct ElectionRecordBody {
-    /// Every encrypted ballot prepared in the election (whether cast or challenged)
-    all_ballots: Vec<BallotEncrypted>,
+    /// Guardian public keys including commitments and proofs of knowledge
+    guardian_public_keys: Vec<GuardianPublicKey>,
+
+    /// Every encrypted ballot prepared in the election (whether cast or challenged) together
+    /// with its weight used in the tally.
+    all_ballots: Vec<(BallotEncrypted, FieldElement)>,
+
+    /// Encrypted tallies of each option
+    encrypted_tallies: BTreeMap<ContestIndex, Vec<Ciphertext>>,
+
+    /// Decrypted tallies with proofs of correct decryption
+    decrypted_tallies: BTreeMap<ContestIndex, Vec<VerifiableDecryption>>,
 
     /// Every challenged ballot
     // challenged_ballots: Vec<BallotSelections>,
 
-    /// Tally of all cast ballots
-
-    /// Ordered lists of ballots encrypted by each device
-    ballots_by_device: HashMap<String, String>,
+    /// Ordered lists of ballots encrypted by each device. The values are indiced of the `all_ballots`
+    /// vector.
+    ballots_by_device: HashMap<String, usize>,
 }
 #[allow(dead_code)]
```

Comment:
Is this still needed?

Reply:
I believe so. According to the specification, the election record should contain "ordered lists of the ballots encrypted by each device".

Reply:
Ah, of course. I think it was added because the private fields in the ElectionRecord where never used. Making them public now.